versioned/persist/serialize/src/main/java/org/projectnessie/versioned/persist/serialize/ProtoSerialization.java

```diff
@@ -26,12 +26,53 @@
 import org.projectnessie.versioned.persist.adapter.ImmutableCommitLogEntry;
 import org.projectnessie.versioned.persist.adapter.ImmutableKeyList;
 import org.projectnessie.versioned.persist.adapter.ImmutableRefLog;
+import org.projectnessie.versioned.persist.adapter.ImmutableRepoDescription;
 import org.projectnessie.versioned.persist.adapter.KeyList;
 import org.projectnessie.versioned.persist.adapter.KeyWithBytes;
 import org.projectnessie.versioned.persist.adapter.KeyWithType;
 import org.projectnessie.versioned.persist.adapter.RefLog;
+import org.projectnessie.versioned.persist.adapter.RepoDescription;
 
 public class ProtoSerialization {
+
+  public static AdapterTypes.RepoProps toProto(RepoDescription repoDescription) {
+    AdapterTypes.RepoProps.Builder proto =
+        AdapterTypes.RepoProps.newBuilder().setRepoVersion(repoDescription.getRepoVersion());
+    repoDescription.getProperties().forEach(proto::putProperties);
```

Comment:
The declaration is of `getProperties` implies a simple `Map`... Can we be sure that the order of entries is stable (which is assumed by database update logic)?

It is likely stable within one particular JVM instance, but might change across environments, I guess...

`TestSerialization.java` does not appear to validate this (unless I missed it).

Reply:
Great catch! Right, there's any guarantee that a (plain) `Map` has a stable order - it's not even guaranteed for e.g. a `HashMap` within the same JVM (unless a bunch of things are or behave exactly the same...).
protobuf 3 defines a map entry as a list of implicit elements - but that list is just the result of some `.entrySet()`.

Fix is easy: redefine the proto `map<string,string>` as a `repeated Entry`.

I'll post a follow-up PR to fix the same issue in `GlobalStatePointer`.